lib/src/react16_suggestors/react_style_maps_updater.dart

```diff
@@ -0,0 +1,263 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:analyzer/analyzer.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:codemod/codemod.dart';
+import 'package:over_react_codemod/src/react16_suggestors/constants.dart';
+
+import './react16_utilities.dart';
+
+/// Suggestor that updates to React 16's StyleMap standard.
+///
+/// Specifically, this suggestor looks for instances where a simple unitless
+/// string literal is passed to a CSS property. React 16 specifies that such
+/// cases should be a num instead of a string.
+class ReactStyleMapsUpdater extends GeneralizingAstVisitor
+    with AstVisitingSuggestorMixin
+    implements Suggestor {
+  @override
+  visitCascadeExpression(CascadeExpression node) {
+    super.visitCascadeExpression(node);
+
+    for (Expression cascade in node.cascadeSections) {
+      if (!hasValidationComment(node, sourceFile)) {
+        if (cascade.toSource().contains('style') &&
+            !cascade.toSource().contains('setProperty')) {
+          /// A style map, method invocation, or a variable
+          dynamic stylesObject = getStyles(cascade);
+
+          /// CSS properties that were modified by the script or need to be
+          /// manually checked
+          List<String> listOfVariables = [];
+
+          // Variables that affect the writing of the comment that goes above
+          // the style prop. There are different messages patched based on
+          // these booleans.
+          bool styleMapContainsAVariable = false;
+          bool isAVariable = false;
+          bool isAFunction = false;
+          bool isOther = false;
+          bool isForCustomProps =
+              cascade.toSource().toLowerCase().contains('props');
+
+          if (stylesObject is MapLiteral) {
+            stylesObject.entries.forEach((MapLiteralEntry cssPropertyRow) {
+              final propertyKey = cssPropertyRow.key;
+              String originalCssPropertyKey = propertyKey.toSource();
+              String cleanedCssPropertyKey = cleanString(propertyKey);
+
+              final cssPropertyValue = cssPropertyRow.value;
+              String cleanedCssPropertyValue = cleanString(cssPropertyValue);
+
+              /// Marks the current CSS property as containing a variable.
+              ///
+              /// Used to keep track of what properties in the style map
+              /// contain variables.
+              void flagAsVariable() {
+                if (!listOfVariables.contains(cleanedCssPropertyKey)) {
+                  listOfVariables.add(cleanedCssPropertyKey);
+                }
+
+                if (!styleMapContainsAVariable) {
+                  styleMapContainsAVariable = true;
+                }
+              }
+
+              num end = cssPropertyRow.end;
```

Comment:
#nit `MapLiteralEntry.end` is actually an `int`... but at any rate - the left side type on this is superfluous since the analyzer can infer it as `int`.

```suggestion
              var end = cssPropertyRow.end;
```